Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
Maybe you have several entity managers? I think there is indeed some more debugging to do. If you think it is the same entity manager, try to add breakpoints to methods that can make the entity become unmanaged (`clean` or `detach`, for instance). Also, maybe try the `$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)` earlier in the request to see if it works at one point, then suddenly doesn't.